win32ss/user/ntuser/winpos.c

```diff
@@ -3709,35 +3709,41 @@ NtUserSetWindowPlacement(HWND hWnd,
    TRACE("Enter NtUserSetWindowPlacement\n");
    UserEnterExclusive();
 
-   if (!(Wnd = UserGetWindowObject(hWnd)) ||
-        UserIsDesktopWindow(Wnd) || UserIsMessageWindow(Wnd))
-   {
-      goto Exit; // Return FALSE
-   }
+    _SEH2_TRY
+    {
+        ProbeForRead(lpwndpl, sizeof(WINDOWPLACEMENT), 1);
+        Safepl = *lpwndpl;
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        SetLastNtError(_SEH2_GetExceptionCode());
+        _SEH2_YIELD(goto Exit); // Return FALSE
+    }
+    _SEH2_END
 
-   _SEH2_TRY
-   {
-      ProbeForRead(lpwndpl, sizeof(WINDOWPLACEMENT), 1);
-      RtlCopyMemory(&Safepl, lpwndpl, sizeof(WINDOWPLACEMENT));
-   }
-   _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
-   {
-      SetLastNtError(_SEH2_GetExceptionCode());
-      _SEH2_YIELD(goto Exit); // Return FALSE
-   }
-   _SEH2_END
+    /* Backwards-compatibility: NT3.x doesn't check the length */
+    if (Safepl.length == sizeof(WINDOWPLACEMENT) ||
+        LOWORD(gptiCurrent->dwExpWinVer) < WINVER_WINNT4)
+    {
+        ERR("0x%lX, 0x%lX\n", gptiCurrent->dwExpWinVer, WINVER_WINNT4);
+        Flags = PLACE_MAX | PLACE_RECT;
+        if (Safepl.flags & WPF_SETMINPOSITION)
+            Flags |= PLACE_MIN;
```

Comment:
Isn't this case also going to call `IntSetWindowPlacement` too?
Because if so, in the case `Safepl.length` is `!= sizeof(WINDOWPLACEMENT)`, it seems that that funtion will make the call fail as well, see https://github.com/reactos/reactos/pull/7096/files#diff-a68d320e37c41490286da8b9fa6aa57411bc1f4630f67eb9506747558a73ad14R563 .
I think you could just unconditionally patch `Safepl.length` to `sizeof(WINDOWPLACEMENT)` in this backwards-compatibility case, since `Safepl` is a local structure, and the contents of `Safepl` aren't returned back to the caller afterwards.

Reply:
Fixed.